### ISSUE TITLE
[Workflow] Workflow actor state should be saved with a TTL of -1

### ISSUE THREAD

**Issue by cgillum** [dapr/dapr#6475]
## In what area(s)?

> /area runtime

## What version of Dapr?

> 1.11

## Expected Behavior

The Dapr Workflow engine should save actor state with a TTL of -1 to ensure any global TTL configuration doesn't cause workflow state to be lost, even if the actor state store has a default TTL specified.

Note that there is some discussion about whether or not TTLs should _ever_ be configured for actor state stores here: https://github.com/dapr/dapr/issues/3658. Either way, the internal workflow actors should defend against this.

## Actual Behavior

Workflow actor state is saved without specifying any TTL, making it vulnerable to state store configurations with TTLs configured.

## Steps to Reproduce the Problem

1. Configure an actor state store with a short TTL
2. Run a long-running workflow that will exceed the above TTL

## Release Note

RELEASE NOTE: **FIX** Prevent actor TTL configuration from disrupting workflows

**Comment by ItalyPaleAle** [comment]
The default behavior of Dapr components is to not set a TTL if the TTL is unset.

**Comment by cgillum** [comment]
@clintsinger or @olitomlinson would you be able to comment further on this issue? This was originally discussed on Discord, and my understanding was that there could be a default/global TTL configuration placed on state stores. I didn't actually do the work to verify this, though.

**Comment by olitomlinson** [comment]
@cgillum 

I think the problem here is that TTLs can be set completely external to Dapr, depending on the chosen state technology

I.e. in CosmosDb, you can set a default TTL which applies to any document in the container. So, as @berndverst suggests, setting ttl to `-1` will override any default TTL behaviours applied outside of Dapr.

**Comment by ItalyPaleAle** [comment]
@olitomlinson yes, in the case of Cosmos DB you can set a default TTL and in that case you must set -1 to override it. I do not believe any other database we support has the same behavior.

IMHO this is not a Workflows issue but rather we just need to document that people shouldn't enable TTLs with a default value in Cosmos DB.

**Comment by cgillum** [comment]
Marking as P2 since this may be a corner case. There's also discussion of excluding Azure Cosmos DB from the list of supported workflow state stores, and it seems that may be the only state store potentially impacted.
